pkg/prometheus/metrics.go

```diff
@@ -15,6 +15,17 @@ const (
 	ErrorResult   = "error"
 )
 
+type Provider string
+
+const (
+	ProviderGitHub    Provider = "github"
+	ProviderGitLab             = "gitlab"
+	ProviderGoogle             = "google"
+	ProviderOpenID             = "openid"
+	ProviderBasicAuth          = "basicauth"
+	ProviderKeystone           = "keystone"
```

Comment:
where's LDAP?

Reply:
indeed, where is it ... let me add support for that one

Reply:
oh i know, we're using ldapclient from library-go and don't have access to the underlying transport.

Reply:
Implementation: https://github.com/openshift/oauth-server/blob/2d5618f7908be0006a239905c60af5f009cc9492/pkg/oauthserver/auth.go#L539-L554

Necessary library-go change: https://github.com/openshift/library-go/pull/1278